### PR TITLE
ref #261: Use DoctrineBundle

### PR DESCRIPTION
**Pull request by umulmrum**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#261
| License       | MIT

### CODE REVIEW

File: UPGRADE-6.3.md [hunk 1]
```diff
@@ -72,11 +107,18 @@ is recommended (although this tool may not find database-related deprecations).
 
 ## Services
 
-None.
+- chameleon_system_core.pdo
```
Comment (UlrichKu) [inline UPGRADE-6.3.md]: Also chameleon_system_core.sessionPdo?

The service "database_connection" is available from Doctrine now?

File: UPGRADE-6.3.md [hunk 2]
```diff
@@ -45,6 +45,41 @@ _errors:
...
+```
+
+The parameters should already be defined.
+
+Please note that the parameter `chameleon_system_core.pdo.enable_mysql_compression` no longer works. To use compression,
+add the configuration value `doctrine: options: 1006: 1`.
+
+The DoctrineBundle provides a profiler in the Web Debug Toolbar. Therefore the Chameleon database profiler is now
+disabled by default. To enable it again, set the configuration key
+`chameleon_system_debug: database_profiler: enabled: true`. Note that the `backtrace_enabled` and `backtrace_limit` keys
```
Comment (UlrichKu) [inline UPGRADE-6.3.md]: I see that there is a legacy config handling for this.

What is the upgrade path here? Will this be documented in the upgrade documentation as a must?

File: UPGRADE-6.3.md [hunk 3]
```diff
@@ -45,6 +45,41 @@ _errors:
 The system now uses Twig 2.x. Please have a look at the Twig changelog for required adjustments, but major problems are
 not expected.
 
+## DoctrineBundle
+
+We now use the DoctrineBundle. While we do not use many of its features yet (especially no ORM mapping), initializing
```
[the remark this answers is not in the document]
Comment (umulmrum) [reply UPGRADE-6.3.md]: This is part of the following sentence.

File: UPGRADE-6.3.md [hunk 4]
```diff
@@ -45,6 +45,41 @@ _errors:
...
+      collate: 'utf8_unicode_ci'
+```
+
+The parameters should already be defined.
+
+Please note that the parameter `chameleon_system_core.pdo.enable_mysql_compression` no longer works. To use compression,
+add the configuration value `doctrine: options: 1006: 1`.
+
+The DoctrineBundle provides a profiler in the Web Debug Toolbar. Therefore the Chameleon database profiler is now
+disabled by default. To enable it again, set the configuration key
```
[the remark this answers is not in the document]
Comment (umulmrum) [reply UPGRADE-6.3.md]: Added some short explanation.

File: UPGRADE-6.3.md (hunk 1 above)
Comment (umulmrum) [reply UPGRADE-6.3.md]: Nope (is still required, but is now initialized differently) and yes.

File: UPGRADE-6.3.md (hunk 2 above)
Comment (umulmrum) [reply UPGRADE-6.3.md]: Added further explanation.

File: UPGRADE-6.3.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (UlrichKu) [reply UPGRADE-6.3.md]: Only reason I could think of: performance impact.

But is backtrace "on" currently?

**Comment by UlrichKu** [comment]
Lots of unrelated pretty ci complaints?

File: UPGRADE-6.3.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (umulmrum) [reply UPGRADE-6.3.md]: OK, I now enabled backtrace by default if the profiler itself is enabled. This should be the most common usage pattern.

**Comment by umulmrum** [comment]
PrettyCI / PHP-CS-Fixer seem to switch order of type annotations now - will investigate separately, as it has nothing to do with this PR.

**Comment by UlrichKu** [comment]
And now a conflict..

**Comment by umulmrum** [comment]
Solved :-)

**Pull request merged**